sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/ChangeFeedProcessorContext.java

```diff
@@ -25,4 +25,11 @@ public interface ChangeFeedProcessorContext {
      * */
     @Beta(value = Beta.SinceVersion.V4_51_0, warningText = Beta.PREVIEW_SUBJECT_TO_CHANGE_WARNING)
     String getLeaseToken();
+
+    /**
+     * Get the diagnostics from the underlying feed response.
+     *
+     * @return The diagnostics object.
+     */
+    CosmosDiagnostics getDiagnostics();
```

Comment:
Maybe we can introduce this as a Beta API as well - following the other APIs in this class. Open to discussion though given this is a very straightforward API.

Reply:
Added to beta